src/main/scala/cognite/spark/v1/SdkV1Relation.scala

```diff
@@ -123,14 +116,37 @@ abstract class SdkV1Relation[A <: Product, I](config: RelationConfig, shortName:
       T <: UpdateByExternalId[R, U, IO] with Create[R, C, IO]](
       existingExternalIds: Seq[String],
       resourceCreates: Seq[C],
-      resource: T)(implicit transform: Transformer[C, U]): IO[Unit] = {
+      resource: T,
+      doUpsert: Boolean)(implicit transform: Transformer[C, U]): IO[Unit] = {
     val (resourcesToUpdate, resourcesToCreate) = resourceCreates.partition(
       p => if (p.externalId.isEmpty) { false } else { existingExternalIds.contains(p.externalId.get) }
     )
-    val create = if (resourcesToCreate.isEmpty) { IO.unit } else {
-      resource.create(resourcesToCreate)
+    val create = if (resourcesToCreate.isEmpty) {
+      IO.unit
+    } else {
+      resource
+        .create(resourcesToCreate)
+        .flatTap { _ =>
+          IO {
+            if (config.collectMetrics) {
+              itemsCreated.inc(resourcesToCreate.size)
+            }
+          }
+        }
+        .handleErrorWith {
+          case e: CdpApiException if doUpsert && e.code == 409 =>
+            val existingExternalIds =
+              e.duplicated.get.map(j => j("externalId").get.asString.get)
+            upsertAfterConflict[R, U, C, T](
+              existingExternalIds,
+              resourcesToCreate,
+              resource,
+              doUpsert = false)
+        }
     }
-    val update = if (resourcesToUpdate.isEmpty) { IO.unit } else {
+    val update = if (resourcesToUpdate.isEmpty) {
+      IO.unit
+    } else {
       resource.updateByExternalId(
         resourcesToUpdate
           .map(u => u.externalId.get -> u.into[U].withFieldComputed(_.externalId, _ => None).transform)
```

Comment:
Shouldn't we be reporting metrics for updates here? I thought that was one of the main points of this PR?